org-roam-buffer.el

```diff
@@ -0,0 +1,266 @@
+               (eq 'visible (org-roam-buffer--visibility))
+               (buffer-local-value 'buffer-file-truename buffer))
+      (setq org-roam-buffer--current buffer)
+      (org-roam-buffer-update))))
+
+;;;; Toggling the org-roam buffer
+(defalias      'org-roam--current-visibility 'org-roam-buffer--visibility)
+(make-obsolete 'org-roam--current-visibility 'org-roam-buffer--visibility "2020/04/06")
+(define-inline org-roam-buffer--visibility ()
+  "Return whether the current visibility state of the org-roam buffer.
+Valid states are 'visible, 'exists and 'none."
+  (declare (side-effect-free t))
+  (inline-quote
+   (cond
+    ((get-buffer-window org-roam-buffer) 'visible)
+    ((get-buffer org-roam-buffer) 'exists)
+    (t 'none))))
+
+(defalias      'org-roam--set-width 'org-roam-buffer--set-width)
+(make-obsolete 'org-roam--set-width 'org-roam-buffer--set-width "2020/04/06")
+(defun org-roam-buffer--set-width (width)
+  "Set the width of `org-roam-buffer' to `WIDTH'."
+  (unless (one-window-p)
+    (let ((window-size-fixed)
+          (w (max width window-min-width)))
+      (cond
+       ((> (window-width) w)
+        (shrink-window-horizontally  (- (window-width) w)))
+       ((< (window-width) w)
+        (enlarge-window-horizontally (- w (window-width))))))))
+
+(defalias      'org-roam--set-height 'org-roam-buffer--set-height)
+(make-obsolete 'org-roam--set-height 'org-roam-buffer--set-height "2020/04/06")
+(defun org-roam-buffer--set-height (height)
+  "Set the height of `org-roam-buffer' to `HEIGHT'."
+  (unless (one-window-p)
+    (let ((window-size-fixed)
+          (h (max height window-min-height)))
+      (cond
+       ((> (window-height) h)
+        (shrink-window  (- (window-height) h)))
+       ((< (window-height) h)
+        (enlarge-window (- h (window-height))))))))
+
+(defalias      'org-roam--set-up-buffer 'org-roam-buffer--get-create)
+(make-obsolete 'org-roam--set-up-buffer 'org-roam-buffer--get-create "2020/04/06")
+(defun org-roam-buffer--get-create ()
+  "Set up the `org-roam' buffer at `org-roam-buffer-position'."
+  (let ((window (get-buffer-window))
+        (position
+         (if (member org-roam-buffer-position '(right left top bottom))
+             org-roam-buffer-position
+           (let ((text-quoting-style 'grave))
+             (lwarn '(org-roam) :error
+                    "Invalid org-roam-buffer-position: %s. Defaulting to \\='right"
+                    org-roam-buffer-position))
+           'right)))
+    (-> (get-buffer-create org-roam-buffer)
+        (display-buffer-in-side-window
+         `((side . ,position)))
+        (select-window))
+    (pcase position
+      ((or 'right 'left)
+       (org-roam-buffer--set-width  (round (* (frame-width)  org-roam-buffer-width))))
+      ((or 'top  'bottom)
+       (org-roam-buffer--set-height (round (* (frame-height) org-roam-buffer-height)))))
+    (select-window window)))
+
+(defun org-roam-buffer-show ()
```

Comment:
This function also hides the buffer if it is shown, so maybe both the docstring and function name needs to change.